core/src/main/java/com/linecorp/armeria/server/ServiceNaming.java

```diff
@@ -51,10 +51,13 @@ static ServiceNaming of(String defaultServiceName) {
     static ServiceNaming fullTypeName() {
         return ctx -> {
             final RpcRequest rpcReq = ctx.rpcRequest();
+            final String serviceName;
             if (rpcReq != null) {
-                return rpcReq.serviceName();
+                serviceName = rpcReq.serviceName();
+            } else {
+                serviceName = ServiceNamingUtil.fullTypeHttpServiceName(ctx.config().service());
             }
-            return ServiceNamingUtil.fullTypeHttpServiceName(ctx.config().service());
+            return ServiceNamingUtil.trimTrailingDollarSigns(serviceName);
```

Comment:
Could you apply `trimTrailingDollarSigns` for the HTTP services? Thrift and gRPC compiler does not allow a service name to ends with `$`.

Reply:
Discussed about classpath confliction. It is too tricky. Understood the intention.